lib/reek/smells/unneeded_suppression.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Reek
+  module Smells
+    #
+    # An unneeded suppression is a Reek's comment disabling a unexistent smell.
+    #
+    # See {file:docs/Unneeded-Suppression.md} for details.
+    class UnneededSuppression
+      def initialize(context, smell_detector)
+        @context = context
+        @smell_detector = smell_detector
+      end
+
+      def sniff
```

Comment:
Again, we're trying really hard to shoehorn this into a detector-style thing. Maybe that's just not the right approach. Please see my comment below as well.